woocommerce/woocommerce-gutenberg-products-block
# Install prettier and reformat files

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/935#discussion_r320796548.

This PR installs `prettier` with the same config from [wp-calypso](https://github.com/Automattic/wp-calypso/blob/master/.prettierrc), refactors all files and adds `prettier` to the pre-commit hooks.

It includes many changes, but I split everything in commits, so reviewing the first (60ea97368109a29e48112111c2e9cb7cd696a1eb) and last one (5854ddc8685f5357807631a904aba23907c0282e) should be enough. The one in the middle was done automatically after running `npm run reformat-files`.

### How to test the changes in this Pull Request:

1. Run `npm start` and verify it stills builds.
2. Create a post and verify blocks are still available and can be added without JS errors.
3. Add an error to a JS file. For example, in [`assets/js/hocs/with-attributes.js`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/hocs/with-attributes.js) make this change:
```diff
-this.state = {
-  attributes: [],
-  error: null,
-  expandedAttribute: null,
-  loading: false,
-  termsList: {},
-  termsLoading: false,
-};
+this.state = {attributes: [], error: null, expandedAttribute: null, loading: false, termsList: {}, termsLoading: false};
```
4. Run `git commit -m "Test"` and verify the change is undone before the commit is made.

### Changelog

> Installed `prettier` so code is formatted automatically.
